### PR TITLE
[ASCII-938] Deprecate `ipc_address` in favor of `cmd_host`

### PR DESCRIPTION
**Pull request by pgimalac**
### What does this PR do?

Deprecate `ipc_address` in favor of `cmd_host`.

### Motivation

Current config names for the existing API (`ipc_address`:`cmd_port`) are confusing.
As a new API server will be added to the agent soon, having clearer names for the existing one would be nice.

### Describe how to test/QA your changes

Check that 
- `ipc_address` can still be used (with text config and environment variable)
- `ipc_address` takes precedence over `cmd_host`
- `cmd_host` is used when `ipc_address` is not provided

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/env/ipc_address.go [hunk 1]
```diff
@@ -38,9 +39,19 @@ func IsLocalAddress(address string) (string, error) {
 
 // GetIPCAddress returns the IPC address or an error if the address is not local
 func GetIPCAddress(cfg pkgconfigmodel.Reader) (string, error) {
-	address, err := IsLocalAddress(cfg.GetString("ipc_address"))
+	var key string
+	// ipc_address is deprecated in favor of cmd_host, but we still need to support it
+	// if it is set, use it, otherwise use cmd_host
+	if cfg.IsSet("ipc_address") {
```
Comment (sgnn7) [inline pkg/config/env/ipc_address.go:45]: Could we add a cipule of tests around this logic?

**Review by sgnn7 — APPROVED** [review]
One comment but generally LGTM. Could we also add a release note too?

**Review by buraizu — APPROVED** [review]

File: pkg/config/env/ipc_address.go (hunk 1 above)
Comment (pgimalac) [reply pkg/config/env/ipc_address.go:45]: Done

**Review by ogaca-dd — APPROVED** [review]

**Review by KevinFairise2 — APPROVED** [review]
Looks good for AP owned files

File: pkg/config/env/ipc_address.go (hunk 1 above)
Comment (AliDatadog) [inline pkg/config/env/ipc_address.go:45]: What if someone sets both ?

Comment (pgimalac) [reply pkg/config/env/ipc_address.go:45]: The old one takes precedence

Comment (AliDatadog) [reply pkg/config/env/ipc_address.go:45]: Is there a reason for that ? TBH I would assume that the newest would be used.

Comment (pgimalac) [reply pkg/config/env/ipc_address.go:45]: It's just backward compatibility, if someone used that key then the behavior is the same

**Review by AliDatadog — APPROVED** [review]

**Review by mfpierre — APPROVED** [review]

**Review by robertjli — APPROVED** [review]

**Pull request merged**